sys/androidmedia/gstamcvideodechybris.c

```diff
@@ -1872,8 +1874,7 @@ gst_amc_video_dec_handle_frame (GstVideoDecoder * decoder,
       switch (idx) {
         case INFO_TRY_AGAIN_LATER:
           GST_DEBUG_OBJECT (self, "Dequeueing input buffer timed out");
-          continue;             /* next try */
-          break;
+          goto dequeue_error;
```

Comment:
Agreed, let's keep it as it is for now. If we ever end up having problems with this new direction, we can keep your alternate idea in mind to try.